### ISSUE TITLE
[MSVC] Object file size cannot exceed 4 GB

### ISSUE THREAD

**Issue by christianbrugger** [stephenberry/glaze#1448]
With the newest release of MSVC 17.12 / 19.42.34433.0 the object file size seems to explode. With Version 17.11 / 19.41 the object has less than 10 MB. Something in the new MSVC makes glaze explode. Here is the log:

```
FAILED: src/core/CMakeFiles/ls_core_lib.dir/serialize_detail.cpp.obj C:
\PROGRA~1\MICROS~2\2022\ENTERP~1\VC\Tools\MSVC\1442~1.344\bin\Hostx64\x64\cl.exe  /nologo /TP -external:ID:
\a\logiksim\logiksim\external\glaze\include -external:W0 /DWIN32 /D_WINDOWS /EHsc /Ob0 
/Od /RTC1 -std:c++20 -MDd -Z7 /arch:AVX2 /diagnostics:caret /utf-8 /permissive- 
-DBL_STATIC /Zc:preprocessor /Zc:lambda /external:anglebrackets /external:W0 /W4 /WX /YuD:/a/logiksim/
logiksim/build/win-msvc-debug/src/core/CMakeFiles/ls_core_lib.dir/cmake_pch.hxx /FpD:/a/logiksim/logiksim/
build/win-msvc-debug/src/core/CMakeFiles/ls_core_lib.dir/./cmake_pch.cxx.pch /FID:/a/logiksim/logiksim/build/
win-msvc-debug/src/core/CMakeFiles/ls_core_lib.dir/cmake_pch.hxx /showIncludes 
/Fosrc\core\CMakeFiles\ls_core_lib.dir\serialize_detail.cpp.obj 
/Fdsrc\core\CMakeFiles\ls_core_lib.dir\ls_core_lib.pdb /FS 
-c D:\a\logiksim\logiksim\src\core\serialize_detail.cpp

D:\a\logiksim\logiksim\src\core\serialize_detail.cpp : 
fatal error C1605: compiler limit: object file size cannot exceed 4 GB
ninja: build stopped: subcommand failed.
```

Link to my failed CI job: https://github.com/christianbrugger/logiksim/actions/runs/11875447310/job/33092660483
This happens when building glaze: https://github.com/stephenberry/glaze/actions/runs/11935224664/job/33266080750

I am sure you guys are aware of the issue. 
How is the status so far. Does this need to be fixed in the MSVC compiler. Or will this be fixable in glaze?

**Comment by stephenberry** [comment]
Yes, this is an MSVC compiler bug, which I reported and they fixed in less than 24 hours. But, we're now waiting on the next incremental release. This seems only to occur with the function `glz::write_json_schema`, so it doesn't affect all of Glaze.

Here is the bug report: https://developercommunity.visualstudio.com/t/Memory-Explosion:-compiler-limit:-objec/10795558

**Comment by arturbac** [comment]
I ran onto this issue with [simple_enum tests](https://github.com/arturbac/simple_enum/pull/16), tired to fix it without luck ..
```cmake
  target_compile_options(glaze_enum_name_ut PRIVATE 
    "/bigobj"
    "/Zm500"  # Increased memory allocation
    "/Os"     # Favor small code size
    )
  target_link_options(glaze_enum_name_ut PRIVATE "/LARGEADDRESSAWARE")
  ```

**Comment by stephenberry** [comment]
Yeah, I think this was due to a memory leak in the MSVC compiler, which they fixed, but we need to wait until the next incremental version to be released.

**Comment by Curve** [comment]
I'm also encountering this issue when using a simple struct with explicit metadata with a `skip` member (Not using json_schema)

**Comment by stephenberry** [comment]
This is an MSVC compiler bug:  https://developercommunity.visualstudio.com/t/Memory-Explosion:-compiler-limit:-objec/10795558

We're awaiting the next MSVC minor release, which is usually every few weeks. They fixed the bug internally, but have yet to release it. You could go back to a previous version of MSVC. They didn't give any indication of what went wrong, so I wouldn't know of any work arounds.

**Comment by Curve** [comment]
> This is an MSVC compiler bug: https://developercommunity.visualstudio.com/t/Memory-Explosion:-compiler-limit:-objec/10795558
> 
> We're awaiting the next MSVC minor release, which is usually every few weeks. They fixed the bug internally, but have yet to release it. You could go back to a previous version of MSVC. They didn't give any indication of what went wrong, so I wouldn't know of any work arounds.

Thanks for the swift response! 

> which is usually every few weeks 

Good to know, hope it'll be released soon ^^

**Comment by stephenberry** [comment]
I checked the MSVC 17.12.3 release from December 3rd and looked at their notes and the fix has not yet been released. Still waiting...

**Comment by dbraeckelmann** [comment]
MSVC 17.13 Preview fixes this problem for me, seems like they might not include it in an incremental 17.12 release after all... -.-

**Comment by helmesjo** [comment]
> MSVC 17.13 Preview fixes this problem for me, seems like they might not include it in an incremental 17.12 release after all... -.-

Interesting... For me it still happens with `17.13.2` (with `/bigobj`).

**Comment by qudix** [comment]
The feedback bot on the relevant [developercommunity](https://developercommunity.visualstudio.com/t/Memory-Explosion:-compiler-limit:-objec/10795558) page seems to suggest a fix for this has been released:

> A fix for this issue has been released! Install the most recent release from https://visualstudio.microsoft.com/downloads/. Thank you for providing valuable feedback which has helped improve the product.

I can't test this myself as I don't have a project large enough (using glaze) to observe the problem in action.

**Comment by dbraeckelmann** [comment]
Yes, they apparently referred to 17.13, I just verified that glaze compiles with the release version.

**Comment by stephenberry** [comment]
Yay! I'm so glad we can close this issue. It is `17.13` that has the fix.

**Comment by arturbac** [comment]
is it already available on github CI ?
